spec/PeriodicBackgroundSync-index.bs

```diff
@@ -133,43 +150,47 @@ The user agent should cap the duration and frequency of these events to limit re
-    1. If |timeTillScheduledEventForOrigin| is not null:
-        1. If |timeTillScheduledEventForOrigin| - |delayForOrigin| is greater than or equal to |minIntervalForOrigin|, abort these substeps.
-        1. If |delayForOrigin| is less than or equal to |timeTillScheduledEventForOrigin|, set |delayForOrigin| to |timeTillScheduledEventForOrigin| and abort these substeps.
-        1. If |delayForOrigin| is less than or equal to |timeTillScheduledEventForOrigin| + |minIntervalForOrigin|, set |delayForOrigin| to |timeTillScheduledEventForOrigin| + |minIntervalForOrigin|.
-    1. Let |timeSinceLastPeriodicSync| be null if [=time the last periodicSync event was fired=] is null, else |now| - [=time the last periodicSync event was fired=].
-    1. If |timeSinceLastPeriodicSync| is null, set |timeToFire| to |delayForOrigin| + |now|.
-    1. Else:
-        1. If |timeSinceLastPeriodicSync| is greater than equal to [=minimum interval across origins=], set |timeToFire| to |delayForOrigin| + |now|.
-        1. Else, set |timeTillNextAllowedPeriodicSync| to [=minimum interval across origins=] - |timeSinceLastPeriodicSync|. Set |timeToFire| to the maximum of |delayForOrigin| + |now|, and |timeTillNextAllowedPeriodicSync| + |now|.
-    1. Set the [=periodicsync registration/time to fire=] of |registration| to |timeToFire|.
-1. Else:
-    1. Increment [=periodicsync registration/count of retries=].
-    1. If [=periodicsync registration/count of retries=] is greater than the [=maximum number of retries=] allowed, set [=periodicsync registration/count of retries=] to 0 and follow the steps to [=calculate the time to fire of the first attempt of=] the periodicSync event.
-    1. Else, Set the [=periodicsync registration/time to fire=] of |registration| to |now| + a small back-off that is exponentially proportional to [=periodicsync registration/count of retries=].
-
-## <dfn>Schedule processing</dfn> ## {#schedule-delayed-processing}
-To [=schedule processing=] of a [=periodicSync registration=] |registration|, run the following steps:
-1. Assert: |registration|'s associated [=periodicsync registration/registration state=] is [=pending=].
-1. Schedule [=fire a periodicSync event|firing a periodicSync event=] for |registration| as soon as the the device is online at or after |registration|'s [=periodicsync registration/time to fire=].
 
+    ## <dfn>Calculate a time to fire</dfn> ## {#caculate-time-to-fire}
+    This section describes how a user agent can calculate the [=periodic sync registration/time to fire=] for a [=periodic sync registration=], |registration|. [=periodic sync registration/time to fire=] value MUST be updated in response to various triggers such as registration, unregistration, change in [=online=] status, and completion of an attempt to fire a [=periodicsync event=].
+    Note: User agents MAY also choose to update this value in response to changes in user engagement levels.
```

Comment:
Avoid throwing out "MAY" statements like this, as it isn't clear how the browser would do this in a compliant manner. In one of my previous comments, I sketched out the algorithm below in a way the browser could call whenever it wants to adjust times in response to things like engagement levels. That allows this feature to exist, but it defines *how* browsers should achieve it.

By putting this MAY here, it seems like the browser could disregard the steps below and do whatever it likes.

Reply:
I've removed this.